changelog.d/11333.misc

```diff
@@ -0,0 +1 @@
+Remove legacy code related to depreciated "trust_identity_server_for_password_resets" config flag.
```

Comment:
I think this could perhaps be written more directly as:

```suggestion
Remove deprecated `trust_identity_server_for_password_resets` configuration flag.
```

The extension for the newsfile should probably be `.removal`?